### ISSUE TITLE
Can't create disks > 2GiB

### ISSUE THREAD

**Issue by elFarto** [vmware/go-vcloud-director#204]
https://github.com/vmware/go-vcloud-director/blob/07823bedaca554587356239bfab4f4e4bb2ed072/types/v56/types.go#L2306
The Size field of the Disk struct is currently an int, rather than the int64 the VCloud XSD says it should be, this limits the maximum size of disk you can create to 2GiB.

**Comment by frenchtoasters** [comment]
> https://github.com/vmware/go-vcloud-director/blob/07823bedaca554587356239bfab4f4e4bb2ed072/types/v56/types.go#L2306
> 
> 
> The Size field of the Disk struct is currently an int, rather than the int64 the VCloud XSD says it should be, this limits the maximum size of disk you can create to 2GiB.

Here is a link to the documentation https://pubs.vmware.com/vca/topic/com.vmware.vcloud.api.reference.doc_56/doc/types/DiskType.html

**Comment by dataclouder** [comment]
Did you actually try creating a disk of more than 2GB?
If yes, what was the error?

According to the documentation the `Size` field should indicate "Size of the disk in MB", thus allowing a theoretical size of about 2 million GB.

**Comment by elFarto** [comment]
Ah..I see my issue, I had assumed that field was in bytes as I had some example code (that for the life of me I can't remember where it came from) that had the size in bytes. I didn't try to allocate a disk bigger than that since I thought it was in bytes and 2GiB+ wouldn't fit in there. So, while that field should be a int64, I think a 2.2PiB maximum should be enough for a while.

**Comment by elFarto** [comment]
I got side tracked with another issue and only recently got back to testing this. It turns out the Size parameter is really in bytes.

The documentation linked above is for vCloud Air, not vCloud Director, here's the actual documentation: https://code.vmware.com/apis/442/vcloud#/doc/doc/types/DiskType.html

I have actually tested it this time, requesting that a disk 10,000,000 bytes big, and that's exactly what showed up in the vcloud interface.
